packages/core/ui-settings/core-ui-settings-server/src/ui_settings_client.ts

```diff
@@ -57,4 +57,8 @@ export interface IUiSettingsClient {
    * Shows whether the uiSetting is a sensitive value. Used by telemetry to not send sensitive values.
    */
   isSensitive: (key: string) => boolean;
+  /**
+   * Returns the validation error message, if one is thrown, when validating the uiSettings value.
+   */
+  getValidationErrorMessage: (key: string, value: unknown) => string | null;
```

Comment:
NIT: maybe just `validate()`?

Also same remark as for the browser-side API, returning something more structured than `string | null` may be preferable.  

Also, to future-proof future usages (e.g async validations such as an index pattern), maybe returning a promise would be better as it would eventually allow later settings to perform async validation?